### PR TITLE
Change gRPC implementation to layer on top of HTTP2 transport

### LINKED ISSUE #8460: Change gRPC implementation to layer on top of HTTP2 transport
**Description:**
This issue is to track changes we need to do to layer gRPC support on top of our HTTP2 transport and get rid of gRPC netty dependencies.

Current Ballerina gRPC implementation depends on grpc-netty implementation to setup http2 connection. we need to use transport-http dependency instead of that.

**Steps to reproduce:**
N/A, this is an improvement.

**Affected Versions:**
N/A, this is an improvement.

**OS, DB, other environment details and versions:**
N/A, this is an improvement.

**Related Issues (optional):**
None

**Suggested Labels (optional):**
improvement

**Suggested Assignees (optional):**
@daneshk

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose
Resolves #8460

## Goals
Current Ballerina gRPC implementation depends on grpc-netty implementation to setup http2 connection. we need to use transport-http dependency instead of grpc-netty. From this fix we remove following grpc dependencies from the ballerina distribution.

> io.grpc:grpc-netty
> io.grpc:grpc-protobuf
> io.grpc:grpc-stub
> io.grpc:grpc-context
> io.grpc:grpc-core
> io.grpc:grpc-protobuf-lite
> com.google.guava:guava(transitive dependency)
> io.opencensus:opencensus-api(transitive dependency)
> io.opencensus:opencensus-contrib-grpc-metrics(transitive dependency)

## Load Test results
This is quick load test done in my local machine(MAC 11) to find out any performance degrade with new implementation, may not be accurate figures.

Test is carried out for simple unary service[1] implemented in ballerina using jmeter java sampler client.
1. https://github.com/ballerina-platform/ballerina-examples/blob/master/examples/grpc-unary-blocking/grpc_unary_blocking.bal

No. of Threads: 1800
Ramp-up period: 30s
Loop count: 500

Results:
screen shot 2018-06-24 at 10 52 12 am

Jconsole observation:
screen shot 2018-06-24 at 10 50 59 am

## Approach
Rewrite gRPC message passing logic on top of HTTP2 transport in wso2/transport-http.

## User stories
Not changes. Change is mainly in backend logic.

## Automation tests
 - Unit tests 
   No
 - Integration tests
   Integration tests are already available. 

## Test environment
MAC, JDK 1.8
 
## Learning
Some code taken from grpc-java(https://github.com/grpc/grpc-java) code.

### CODE REVIEW

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientCall.java [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    }
+
+    /**
+     * Prevent any further processing for the remote call.
+     *
+     * @param message error message
+     * @param cause Throwable
+     */
+    public void cancel(String message, Throwable cause) {
+
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientCall.java]: Extra new line? Check all new methods

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+import static org.ballerinalang.net.grpc.GrpcConstants.MESSAGE_ENCODING;
+import static org.ballerinalang.net.grpc.MessageUtils.readAsString;
+
+/**
+ * Client Connector Listener.
+ */
+public class ClientConnectorListener implements HttpClientConnectorListener {
+
+    private static final Logger log = LoggerFactory.getLogger(ClientConnectorListener.class);
+    ClientCall call;
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java]: Shall we make this private

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        }
+        stateListener.transportReportStatus(transportError, false, transportErrorMetadata);
+    }
+
+    /**
+     * Called by subclasses whenever {@code Headers} are received from the transport.
+     *
+     * @param inboundMessage the incoming message.
+     */
+    protected boolean isValid(InboundMessage inboundMessage) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java]: Shall we make this private?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        }
+        return null;
+    }
+
+    /**
+     * Called by subclasses for the terminal trailer metadata on a stream.
+     *
+     * @param trailers the received terminal trailer metadata
+     */
+    protected void transportTrailersReceived(HttpHeaders trailers) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java]: Private method?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+            if (!listenerClosed) {
+                listenerClosed = true;
+                listener().closed(status, trailers);
+            }
+        }
+
+        @Override
+        public void deframeFailed(Throwable cause) {
+            transportReportStatus(Status.fromThrowable(cause), true, new DefaultHttpHeaders());;
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java]: Extra semicolon?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+        }
+    }
+
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java]: Extra new line

Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java]: Extra new line?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+                            Status.Code.INTERNAL.toStatus()
+                                    .withDescription(String.format("Can't find decompressor for %s", messageEncoding))
+                                    .asRuntimeException());
+                    return;
+                } else if (decompressor != Codec.Identity.NONE) {
+                    setDecompressor(decompressor);
+                }
+            }
+
+            listener().headersRead(headers);
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ClientConnectorListener.java]: Shall we consider reformat the above method's code?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Codec.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        @Override
+        public InputStream decompress(InputStream is) throws IOException {
+            return new GZIPInputStream(is);
+        }
+    }
+
+    /**
+     * The "identity", or "none" codec.
+     */
+    public static final class Identity implements Codec {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Codec.java]: Do we need public static modifier for interface inner classes?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Codec.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+public interface Codec extends Compressor, Decompressor {
+
+    /**
+     * A gzip compressor and decompressor.
+     */
+    public static final class Gzip implements Codec {
+
+        @Override
+        public String getMessageEncoding() {
+            return "gzip";
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Codec.java]: Shall we use a constant?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Codec.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        public static final Codec NONE = new Identity();
+
+        @Override
+        public InputStream decompress(InputStream is) throws IOException {
+            return is;
+        }
+
+        @Override
+        public String getMessageEncoding() {
+            return "identity";
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Codec.java]: Shall we use a constant?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/CompositeContent.java [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+    public final int readInt() {
+        checkReadable(4);
+        int b1 = readUnsignedByte();
+        int b2 = readUnsignedByte();
+        int b3 = readUnsignedByte();
+        int b4 = readUnsignedByte();
+        return (b1 << 24) | (b2 << 16) | (b3 << 8) | b4;
+    }
+
+    protected final void checkReadable(int length) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/CompositeContent.java]: Access can be private?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/CompositeContent.java [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+            int readInternal(ByteBuf buffer, int length) {
+
+                return buffer.readUnsignedByte();
+            }
+        };
+        execute(op, 1);
+        return op.value;
+    }
+
+    public void skipBytes(int length) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/CompositeContent.java]: Unused method?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/CompositeContent.java [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+            public int readInternal(ByteBuf buffer, int length) {
+
+                buffer.skipBytes(length);
+                return 0;
+            }
+        }, length);
+    }
+
+    public void readBytes(final byte[] dest, final int destOffset, int length) {
+
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/CompositeContent.java]: Check for these unnecessary new lines all the places

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/GrpcCallableUnitCallBack.java [hunk 14]
```diff
@@ -61,7 +62,7 @@ public void notifySuccess() {
         // notify success only if response message is empty. Service impl doesn't send empty message. Empty response
         // scenarios handles here.
         if (emptyResponse) {
-            requestSender.onNext(Message.newBuilder("Empty").build());
+            requestSender.onNext((ResponseT) new Message("Empty"));
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/GrpcCallableUnitCallBack.java]: Can we use GrpcConstants.EMPTY_DATATYPE_NAME here

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/InboundMessage.java [hunk 15]
```diff
@@ -0,0 +1,218 @@
...
+    public void setProperty(String key, Object value) {
+        httpCarbonMessage.setProperty(key, value);
+    }
+
+    /**
+     * Remove a property from the underlining CarbonMessage object.
+     *
+     * @param key property key
+     */
+    public void removeProperty(String key) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/InboundMessage.java]: Do we need this unused method?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Message.java [hunk 16]
```diff
@@ -32,39 +35,61 @@
...
     public Map<String, Object> getFields() {
         return fields;
     }
-    
-    @Override
-    public final com.google.protobuf.UnknownFieldSet getUnknownFields() {
-        return this.unknownFields;
+
+    public HttpHeaders getHeaders() {
+
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Message.java]: Remove extra new lines.check all the methods

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Message.java [hunk 17]
```diff
@@ -74,196 +99,173 @@ public Message(
...
+            int tag = input.readTag();
+            if (tag == 0) {
+                done = true;
+            } else if (fields.containsKey(tag)) {
+                Descriptors.FieldDescriptor fieldDescriptor = fields.get(tag);
+                String name = fieldDescriptor.getName();
+                switch (fieldDescriptor.getType().toProto().getNumber()) {
+                    case DescriptorProtos.FieldDescriptorProto.Type.TYPE_DOUBLE_VALUE: {
+                        if (fieldDescriptor.isRepeated()) {
+                            List<Double> messages = new ArrayList<>();
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Message.java]: Shall we move this to a else block?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Message.java [hunk 18]
```diff
@@ -556,26 +578,29 @@ public int getSerializedSize() {
...
-                                        .getNumber(), (MessageLite) message);
+                            Message[] messages = (Message[]) msgObject;
+                            for (Message message : messages) {
+                                size += computeMessageSize(fieldDescriptor, message);
                             }
                         } else {
-                            size += com.google.protobuf.CodedOutputStream.computeMessageSize(fieldDescriptor
-                                    .getNumber(), (MessageLite) fields.get(fieldDescriptor.getName()));
+                            size += computeMessageSize(fieldDescriptor, (Message) fields.get(fieldDescriptor.getName
+                                    ()));
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Message.java]: Shall we reformat this line?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageFramer.java [hunk 19]
```diff
@@ -0,0 +1,377 @@
...
+
+    public void setCompressor(Compressor compressor) {
+        this.compressor = compressor;
+    }
+
+    public void setMessageCompression(boolean enable) {
+        messageCompression = enable;
+    }
+
+    public void setMaxOutboundMessageSize(int maxSize) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageFramer.java]: Unused method?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageFramer.java [hunk 20]
```diff
@@ -0,0 +1,377 @@
...
+    /**
+     * Writes out a payload message.
+     *
+     * @param message message to be written in form of input stream.
+     */
+    public void writePayload(InputStream message) {
+        verifyNotClosed();
+        boolean compressed = messageCompression && compressor != Codec.Identity.NONE;
+        int written = -1;
+        int messageLength = -2;
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageFramer.java]: These variable initializations are redundant.

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageFramer.java [hunk 21]
```diff
@@ -0,0 +1,377 @@
...
+    /**
+     * Write an unserialized/uncompressed message with a known length.
+     */
+    private int writeKnownLengthUncompressed(InputStream message, int messageLength)
+            throws IOException {
+        if (maxOutboundMessageSize >= 0 && messageLength > maxOutboundMessageSize) {
+            throw Status.Code.RESOURCE_EXHAUSTED.toStatus()
+                    .withDescription(
+                            String.format("message too large %d > %d", messageLength, maxOutboundMessageSize))
+                    .asRuntimeException();
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageFramer.java]: Shall we reformat this line

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageFramer.java [hunk 22]
```diff
@@ -0,0 +1,377 @@
...
+        ByteBuffer writeableHeader = ByteBuffer.allocate(HEADER_LENGTH);
+        writeableHeader.put(headerScratch, 0, header.position());
+        if (messageLength == 0) {
+            buffer = writeableHeader;
+            return;
+        }
+        // Write content as default http content.
+        carbonMessage.addHttpContent(new DefaultHttpContent(Unpooled.wrappedBuffer(writeableHeader)));
+        List<ByteBuffer> bufferList = bufferChain.bufferList;
+        for (int i = 0; i < bufferList.size(); i++) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MessageFramer.java]: Shall we use foreach here?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MethodDescriptor.java [hunk 23]
```diff
@@ -0,0 +1,351 @@
...
+     */
+    public interface Marshaller<Message> {
+
+        /**
+         * Produces an {@link InputStream} for given messages.
+         *
+         * @param value to serialize.
+         * @return serialized value as stream of bytes.
+         */
+        public InputStream stream(Message value);
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/MethodDescriptor.java]: We don't need public identifier here for both methods

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ServerConnectorListener.java [hunk 24]
```diff
@@ -0,0 +1,251 @@
...
+import static org.ballerinalang.net.grpc.GrpcConstants.GRPC_MESSAGE_KEY;
+import static org.ballerinalang.net.grpc.GrpcConstants.GRPC_STATUS_KEY;
+
+/**
+ * gRPC connector listener for Ballerina.
+ */
+public class ServerConnectorListener implements HttpConnectorListener {
+
+    private static final Logger log = LoggerFactory.getLogger(ServerConnectorListener.class);
+    private static final PrintStream console = System.out;
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ServerConnectorListener.java]: Unused console ?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ServerMethodDefinition.java [hunk 25]
```diff
@@ -0,0 +1,76 @@
...
+     *
+     * @param method  method descriptor for this method.
+     * @param handler server handler to dispatch calls.
+     * @return a new instance.
+     */
+    public static <ReqT, RespT> ServerMethodDefinition<ReqT, RespT> create(
+            MethodDescriptor<ReqT, RespT> method,
+            ServerCallHandler<ReqT, RespT> handler) {
+
+        return new ServerMethodDefinition<ReqT, RespT>(method, handler);
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ServerMethodDefinition.java]: This can be replaced with ServerMethodDefinition<>(method, handler);

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ServerServiceDefinition.java [hunk 26]
```diff
@@ -0,0 +1,202 @@
...
+
+        private Builder(String serviceName) throws GrpcServerException {
+            if (serviceName == null) {
+                throw new GrpcServerException("Service Name cannot be null");
+            }
+            this.serviceName = serviceName;
+            this.serviceDescriptor = null;
+        }
+
+        private Builder(ServiceDescriptor serviceDescriptor) throws GrpcServerException {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ServerServiceDefinition.java]: Why do we need this constructor?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ServerServiceDefinition.java [hunk 27]
```diff
@@ -0,0 +1,202 @@
...
+         * @param handler handler for incoming calls
+         */
+        public <ReqT, RespT> void addMethod(
+                MethodDescriptor<ReqT, RespT> method, ServerCallHandler<ReqT, RespT> handler) throws
+                GrpcServerException {
+            if (method == null) {
+                throw new GrpcServerException("Service Descriptor cannot be null");
+            }
+            if (handler == null) {
+                throw new GrpcServerException("Service Descriptor cannot be null");
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/ServerServiceDefinition.java]: Shall we have two different error messages for the two cases?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Status.java [hunk 28]
```diff
@@ -0,0 +1,458 @@
...
+        private byte[] valueAscii() {
+            return valueAscii;
+        }
+    }
+
+    // Create the canonical list of Status instances indexed by their code values.
+    private static final List<Status> STATUS_LIST = buildStatusList();
+
+    private static List<Status> buildStatusList() {
+        TreeMap<Integer, Status> canonicalizer = new TreeMap<Integer, Status>();
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Status.java]: This can be changed to new TreeMap<>();

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Status.java [hunk 29]
```diff
@@ -0,0 +1,458 @@
...
+        }
+        return Collections.unmodifiableList(new ArrayList<Status>(canonicalizer.values()));
+    }
+
+    /**
+     * Return a {@link Status} given a canonical error {@link Code} value.
+     */
+    public static Status fromCodeValue(int codeValue) {
+        if (codeValue < 0 || codeValue > STATUS_LIST.size()) {
+            return Status.Code.UNKNOWN.toStatus().withDescription("Unknown code " + codeValue);
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/Status.java]: Shall we use a constant for this?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/listener/ServerCallHandler.java [hunk 30]
```diff
@@ -0,0 +1,313 @@
...
+         * <p>
+         * <p>There will be no further callbacks for the call.
+         */
+        public void onComplete() {
+
+        }
+
+        /**
+         * This indicates that the call is now capable of sending additional messages (via
+         * {@link #sendMessage}) without requiring excessive buffering internally. This event is
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/listener/ServerCallHandler.java]: Is the link sendMessage correct?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/listener/ServerCallHandler.java [hunk 31]
```diff
@@ -0,0 +1,313 @@
...
+
+    /**
+     * Returns BValue object corresponding to the protobuf request message.
+     *
+     * @param requestMessage protobuf request message.
+     * @return b7a message.
+     */
+    private BValue getRequestParameter(Resource resource, RequestT requestMessage, boolean isHeaderRequired) {
+        if (resource == null || resource.getParamDetails() == null || resource.getParamDetails().size() > 3) {
+            throw new RuntimeException("Invalid resource input arguments. arguments must not be greater than three");
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/listener/ServerCallHandler.java]: Isn't this error misleading on the case of resource == null  or resource.getParamDetails() == null  as it is not greater than 3?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/listener/UnaryServerCallHandler.java [hunk 32]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        @Override
+        public void onHalfClose() {
+            if (!canInvoke) {
+                return;
+            }
+            if (request == null) {
+                call.close(
+                        Status.Code.INTERNAL.toStatus().withDescription(MISSING_REQUEST), new DefaultHttpHeaders());
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/listener/UnaryServerCallHandler.java]: Reformat the code

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/calleraction/Send.java [hunk 33]
```diff
@@ -77,27 +78,29 @@ public void execute(Context context) {
...
+                            "connector. Response sender does not exist"))));
         } else {
-            io.grpc.Context previous = msgContext != null ? msgContext.attach() : null;
             try {
                 // If there is no response message like conn -> send(), system doesn't send the message.
                 if (!MessageUtils.isEmptyResponse(outputType)) {
                     Message responseMessage = MessageUtils.generateProtoMessage(responseValue, outputType);
+                    // Update response headers when request headers exists in the context.
+                    HttpHeaders headers = null;
+                    if (headerValues instanceof BStruct) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/calleraction/Send.java]: Instead of instanceof shall we check the type tag?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/client/Complete.java [hunk 34]
```diff
@@ -60,8 +59,8 @@ public void execute(Context context) {
         StreamObserver requestSender = (StreamObserver) connectionStruct.getNativeData(REQUEST_SENDER);
         if (requestSender == null) {
             context.setError(MessageUtils.getConnectorError(context, new StatusRuntimeException(Status
-                    .fromCode(Status.INTERNAL.getCode()).withDescription("Error while initializing connector. " +
-                            "response sender does not exist"))));
+                    .fromCode(Status.Code.INTERNAL.toStatus().getCode()).withDescription("Error while initializing " +
+                            "connector. response sender does not exist"))));
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/client/Complete.java]: Shall we reword this as we don't have connector concept in ballerina now?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/clientendpoint/Init.java [hunk 35]
```diff
@@ -74,60 +70,50 @@
...
-     * Creates a new {@link InetSocketAddress} on localhost that overrides the host with.
-     */
-    private static InetSocketAddress generateSocketAddress(String host, int port) {
+        Struct clientEndpoint = BLangConnectorSPIUtil.getConnectorEndpointStruct(context);
+        // Creating client endpoint with channel as native data.
+        BStruct endpointConfigStruct = (BStruct) context.getRefArgument(1);
+        Struct endpointConfig = BLangConnectorSPIUtil.toStruct(endpointConfigStruct);
+        String urlString = endpointConfig.getStringField(GrpcConstants.CLIENT_ENDPOINT_URL);
+        HttpConnectionManager connectionManager = HttpConnectionManager.getInstance();
+        String scheme;
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/clientendpoint/Init.java]: Shall we move this declaration to the assign statement it self?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/headers/Get.java [hunk 36]
```diff
@@ -52,19 +49,19 @@
...
+        String headerValue = headers != null ? headers.get(headerName) : null;
         if (headerValue != null) {
             context.setReturnValues(new BString(headerValue));
         } else {
             context.setReturnValues();
         }
     }
 
-    private String getHeaderValue(MessageHeaders metadata, String keyName) {
+/*    private String getHeaderValue(MessageHeaders metadata, String keyName) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/headers/Get.java]: Shall we remove this commented code block?

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/headers/GetAll.java [hunk 37]
```diff
@@ -54,15 +51,23 @@
 public class GetAll extends BlockingNativeCallableUnit {
     @Override
     public void execute(Context context) {
+        //TODO: redesign headers support
         String headerName = context.getStringArgument(0);
         BStruct headerValues = (BStruct) context.getRefArgument(0);
-        MessageHeaders metadata = headerValues != null ? (MessageHeaders) headerValues.getNativeData(METADATA_KEY)
+        HttpHeaders headers = headerValues != null ? (HttpHeaders) headerValues.getNativeData(MESSAGE_HEADERS)
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/headers/GetAll.java]: Reformat the line

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/stubs/BlockingStub.java [hunk 38]
```diff
@@ -0,0 +1,183 @@
...
+        } else if (protoType.equalsIgnoreCase("Int32Value") || protoType
+                .equalsIgnoreCase("Int64Value") || protoType
+                .equalsIgnoreCase("UInt32Value") || protoType
+                .equalsIgnoreCase("UInt64Value")) {
+            return BTypes.typeInt;
+        } else if (protoType.equalsIgnoreCase("BoolValue")) {
+            return BTypes.typeBoolean;
+        } else if (protoType.equalsIgnoreCase("StringValue")) {
+            return BTypes.typeString;
+        } else if (protoType.equalsIgnoreCase("BytesValue")) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/stubs/BlockingStub.java]: Consider using constants for these values

File: stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/stubs/NonBlockingStub.java [hunk 39]
```diff
@@ -0,0 +1,207 @@
...
+            return call.isReady();
+        }
+
+        @Override
+        public void setMessageCompression(boolean enable) {
+
+            call.setMessageCompression(enable);
+        }
+
+        public void cancel(@Nullable String message, @Nullable Throwable cause) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/stubs/NonBlockingStub.java]: Unused method?

**Comment by daneshk** [comment]
@anupama-pathirage Thanks for reviewing the PR. I corrected all pointed code issues.
